API/Async.pm

```diff
@@ -274,6 +274,7 @@ sub getFavorites {
 			$cb->($items);
 		},{
 			_ttl => USER_CONTENT_TTL,
+			_nocache => 1,
```

Comment:
`USER_CONTENT_TTL` is five minutes. We should not disable caching as the data would be looked up repeatedly while navigating the menu: each time you drill down a level, the previous levels would be evaluated again. It's just how SlimBrowse works. Therefore I decided to _always_ cache, even if it's only five minutes. This should guarantee that we don't have to do the lookup on each button press, or to be surprised by changing parent menu items.

Reply:
Are you sure this function is used during the descending of the tree? And when you know the timestamp has changed, then you'll request the data again but the _get will return from cache and you'll miss. I'm saying that b/c the gettimestamp disables cache for that reason, but then after we read the wrong data and update incorrectly the timestamp cache. Aren't we in a lockup here? Deezer is different BTW because it provides a checksum of the dataset, so you know with one track query, not cached, that something has changed

Reply:
I mean, if the collection is large, we always get the timestamp, un-cached, even when drilling down. But then, we don't use properly that information because even if the timestamp has changed, we'll then call lookup that will unfortunately use the _get cache, although we explicitly don't want it to do it because we know data has changed.
And if timestamp is unchanged, then lookup is not called.

Reply:
Here is what I've done in Deezer and my understanding of what you did

- There are 2 levels of cache: the result of the HTTP get (short ttl) and the OPML items (long ttl)
- The $lookupSub is only called when we know for sure that we need to build a fresh list of items, so it should not use HTTP get cache
- We check that we have OPML cached items 
- If we don't, we need to HTTP get them
- If we do, then verify that this list has been outdated (checksum) by HTTP get'ting a small set of items (only 4). This must be done w/o HTTP cache because otherwise we don't know if there is an update
- If the small GET tells us there is an update, then call $lookupSub which for sure must be done w/o HTTP get cache as something has changed.

- In your case you have an OPML size limit below which you always call $lookupSub and that might be the difference .
- In my case, I don't have such limit, so I will always verify outdated/checksum **except** if the quantity requested is 1 which means we are drilling down so there is no purpose to verify. In all other cases, I'll verify the checksum by HTTP get'ting 1 item, not cached.

I assumed that you might not agree with the trick of not calling the checksum verification only when quantity is 1, which means you prefer to have $lookupSub use cache. 

The problem I think I saw it, is that, when you are in the case of a large OPML list, you check outdated/checksum, it tells you something has changed, so you do a $lookupSub but because it allows use of cache, it returns the same thing as before but you update the timestamp so next calls you'll think that there is nothing new now and it feels like a deadlock to me because you never get the updates: your timestamp will never be outdated and even if the HTTP get short ttl cache expires, the long OPML list one will prevail. It only works if at least one attempt to detect a change is made *after* the short ttl of HTTP get expired.

Reply:
I'm sorry I had you write all this up... I was about to write back my agreement when I got distracted and forgot to send off the response... The checksum test was a later addition which I didn't take into account in my first response. Nor the local caching inside the lookup routine.

Now... may I still ask you to remove the `_ttl` value, too? It's useless there.

Reply:
No worries and writing always help me clarifying my thoughts. Do you have an opinion on the quantity == 1 means drill thing?

Reply:
I never thought of this level of optimisation. Previously I would simply cache, the "checksum" thing is relatively new. I think in this case I'd go for simple over optimised.